apis/workloads/v1alpha1/priority_webhook.go

```diff
@@ -115,39 +91,45 @@ func (i *injector) Handle(ctx context.Context, req types.Request) (resp types.Re
 		}
 
 		// Catch any Allowed=false responses, as this means we've failed to accept this pod
-		if !resp.Response.Allowed {
+		if !resp.Allowed {
 			errorsTotal.With(labels).Inc()
 		}
 	}(time.Now())
 
 	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: req.AdmissionRequest.Namespace,
+			Name: req.Namespace,
 		},
 	}
 	nsName, _ := client.ObjectKeyFromObject(ns)
 	if err := i.client.Get(ctx, nsName, ns); err != nil {
-		return admission.ErrorResponse(http.StatusBadRequest, err)
+		return admission.Errored(http.StatusBadRequest, err)
 	}
 
 	pod := &corev1.Pod{}
 	if err := i.decoder.Decode(req, pod); err != nil {
-		return admission.ErrorResponse(http.StatusBadRequest, err)
+		return admission.Errored(http.StatusBadRequest, err)
 	}
 
 	priorityClassName, ok := ns.ObjectMeta.Labels[NamespaceLabel]
 	if !ok {
-		logger.Log("event", "pod.skipped", "msg", "no priority label found")
+		logger.Info("skipping pod without priority label", "event", "pod.skipped", "msg", "no priority label found")
 		skipTotal.With(labels).Inc()
-		return admission.PatchResponse(pod, pod)
+		return admission.Allowed("no priority label found")
 	}
 
 	mutateTotal.With(labels).Inc() // we are committed to mutating this pod now
 
-	logger.Log("event", "pod.assign_priority_class", "class", priorityClassName)
+	logger.Info(fmt.Sprintf("pod assigned priority class %s", priorityClassName), "event", "pod.assign_priority_class", "class", priorityClassName)
 	copy := pod.DeepCopy()
 	copy.Spec.PriorityClassName = priorityClassName
 	copy.Spec.Priority = nil
 
-	return admission.PatchResponse(pod, copy)
+	// TODO(jackatbancast): convert using JSON patch operations
```

Comment:
?

Reply:
Patching here would probably be better than creating both objects, marshalling into JSON and then having controller runtime figure out the patches for us.

Using JSON patches directly would probably help us to structure our code in a form that we can more easily reason about.